cluster-autoscaler/cloudprovider/gce/gce_price_model.go

```diff
@@ -0,0 +1,168 @@
+		"n1-highmem-4":   0.0500,
+		"n1-highmem-8":   0.1000,
+		"n1-highmem-16":  0.2000,
+		"n1-highmem-32":  0.4000,
+		"n1-highmem-64":  0.8000,
+		"n1-highcpu-2":   0.0150,
+		"n1-highcpu-4":   0.0300,
+		"n1-highcpu-8":   0.0600,
+		"n1-highcpu-16":  0.1200,
+		"n1-highcpu-32":  0.2400,
+		"n1-highcpu-64":  0.4800,
+	}
+)
+
+// NodePrice returns a price of running the given node for a given period of time.
+// All prices are in USD.
+func (model *GcePriceModel) NodePrice(node *apiv1.Node, startTime time.Time, endTime time.Time) (float64, error) {
+	price := 0.0
+	basePriceFound := false
+	if node.Labels != nil {
+		if machineType, found := node.Labels[metav1.LabelInstanceType]; found {
+			var priceMapToUse map[string]float64
+			if node.Labels[preemptibleLabel] == "true" {
+				priceMapToUse = preemptiblePrices
+			} else {
+				priceMapToUse = instancePrices
+			}
+			if basePricePerHour, found := priceMapToUse[machineType]; found {
+				price = basePricePerHour * getHours(startTime, endTime)
+				basePriceFound = true
+			}
+		}
+	}
+	if basePriceFound == false {
+		price = getBasePrice(node.Status.Capacity, startTime, endTime)
+		if node.Labels != nil && node.Labels[preemptibleLabel] == "true" {
+			price = price * preemptibleDiscount
+		}
+		basePriceFound = true
+	}
+	if !basePriceFound {
+		return 0, fmt.Errorf("failed to calculate price for node %s", node.Name)
+	}
+	price += getAdditionalPrice(node.Status.Capacity, startTime, endTime)
+
+	glog.Warningf("Price of %s is %f", node.Name, price)
+	return price, nil
+}
+
+func getHours(startTime time.Time, endTime time.Time) float64 {
+	minutes := math.Ceil(float64(endTime.Sub(startTime)) / float64(time.Minute))
+	hours := minutes / 60.0
+	return hours
+}
+
+// PodePrice returns a theoretical minimum priece of running a pod for a given
+// period of time on a perfectly matching machine.
+func (model *GcePriceModel) PodPrice(pod *apiv1.Pod, startTime time.Time, endTime time.Time) (float64, error) {
+	price := 0.0
+	for _, container := range pod.Spec.Containers {
+		price += getBasePrice(container.Resources.Requests, startTime, endTime)
+		price += getAdditionalPrice(container.Resources.Requests, startTime, endTime)
+	}
+	return price, nil
+}
+
+func getBasePrice(resources apiv1.ResourceList, startTime time.Time, endTime time.Time) float64 {
+	if len(resources) == 0 {
+		return 0
+	}
+	hours := getHours(startTime, endTime)
+	price := 0.0
+	cpu := resources[apiv1.ResourceCPU]
+	mem := resources[apiv1.ResourceMemory]
+	price += float64(cpu.MilliValue()) / 1000.0 * cpuPricePerHour * hours
+	price += float64(mem.Value()) / gigabyte * memoryPricePerHourPerGb * hours
+	return price
+}
+
+func getAdditionalPrice(resources apiv1.ResourceList, startTime time.Time, endTime time.Time) float64 {
+	if len(resources) == 0 {
+		return 0
+	}
+	hours := getHours(startTime, endTime)
+	price := 0.0
+	gpu := resources[apiv1.ResourceNvidiaGPU]
+	price += float64(gpu.MilliValue()) / 1000.0 * gpuPricePerHour * hours
```

Comment:
Either put this line inside if (based on whether gpu is found in ResourceList), or just declare the variable here.